libraries/c_sdk/standard/mqtt/src/iot_mqtt_network.c

```diff
@@ -387,6 +388,20 @@ static IotMqttError_t _deserializeIncomingPacket( _mqttConnection_t * pMqttConne
             /* Deserialize incoming PUBLISH. */
             status = deserialize( pIncomingPacket );
 
+            if( status == IOT_MQTT_SUCCESS )
+            {
+                contextIndex = _IotMqtt_getContextIndexFromConnection( pMqttConnection );
+
+                if( contextIndex >= 0 )
+                {
+                    IotMutex_Lock( &( connToContext[ contextIndex ].contextMutex ) );
+                    publishStateStatus = MQTT_UpdateStatePublish( &( connToContext[ contextIndex ].context ),
```

Comment:
Please add comments about why this is required

Reply:
Thinking about it, this state update may not be required for incoming publishes. If we are doing it, we would need to update the state after sending a PUBACK for QoS1 publishes.